hibernate-core/src/main/java/org/hibernate/dialect/DB2400Dialect.java

```diff
@@ -21,81 +21,16 @@
  */
 public class DB2400Dialect extends DB2Dialect {
 
-	private static final AbstractLimitHandler LIMIT_HANDLER = new AbstractLimitHandler() {
-		@Override
-		public String processSql(String sql, RowSelection selection) {
-			if ( LimitHelper.hasFirstRow( selection ) ) {
-				//nest the main query in an outer select
-				return "select * from ( select inner2_.*, rownumber() over(order by order of inner2_) as rownumber_ from ( "
-						+ sql + " fetch first " + getMaxOrLimit( selection ) + " rows only ) as inner2_ ) as inner1_ where rownumber_ > "
-						+ selection.getFirstRow() + " order by rownumber_";
-			}
-			return sql + " fetch first " + getMaxOrLimit( selection ) + " rows only";
-		}
-
-		@Override
-		public boolean supportsLimit() {
-			return true;
-		}
-
-		@Override
-		public boolean useMaxForLimit() {
-			return true;
-		}
-
-		@Override
-		public boolean supportsVariableLimit() {
-			return false;
-		}
-	};
-
 	@Override
 	public boolean supportsSequences() {
 		return false;
 	}
 
-	@Override
-	public boolean supportsLimit() {
-		return true;
-	}
-
 	@Override
 	public String getQuerySequencesString() {
 		return null;
 	}
 
-	@Override
-	@SuppressWarnings("deprecation")
-	public boolean supportsLimitOffset() {
-		return false;
-	}
-
-	@Override
-	public boolean useMaxForLimit() {
-		return true;
-	}
-
-	@Override
-	public boolean supportsVariableLimit() {
-		return false;
-	}
-
-	@Override
-	public String getLimitString(String sql, int offset, int limit) {
-		if ( offset > 0 ) {
-			throw new UnsupportedOperationException( "query result offset is not supported" );
-		}
-		if ( limit == 0 ) {
-			return sql;
-		}
-		return sql + " fetch first " + limit + " rows only ";
-	}
-
-	@Override
-	public LimitHandler getLimitHandler() {
-		return LIMIT_HANDLER;
-	}
-
```

Comment:
I removed this lines because the implementation is the same as the superclass. However rereading my commit I discovered I removed a restriction with offset that existed in an old OS version. I reintroduce it in the next commit.

Reply:
I may be wrong but I think also 
	```
public boolean supportsLimitOffset() {	
		return false;	
	}
```
should be reintroduced.

Reply:
Hi, shouldn't it be set to true?

Otherwise offset will always come to zero in the getLimitString method, so it will always execute the query as in the getLimitString method of the DB2400Dialect dialect, without offset.